src/engine-actions/get-chain-transactions.js

```diff
@@ -0,0 +1,194 @@
+ */
+async function getChainTransactions () {
+  const transactions = await getTransactions({ client: this.client })
+
+  if (!Array.isArray(transactions) || !transactions.length) {
+    this.logger.debug('No transactions exist on the engine')
+    return []
+  }
+
+  // Get all channels from the engine (rough)
+  const [
+    { channels: rawChannels },
+    { channels: rawClosedChannels },
+    {
+      pendingOpenChannels: rawPendingOpenChannels,
+      pendingClosingChannels: rawPendingClosingChannels,
+      pendingForceClosingChannels: rawPendingForceClosingChannels,
+      waitingCloseChannels: rawWaitingCloseChannels
+    }
+  ] = await Promise.all([
+    listChannels({ client: this.client }),
+    listClosedChannels({ client: this.client }),
+    listPendingChannels({ client: this.client })
+  ])
+
+  const openChannels = [
+    ...rawChannels,
+    // Pending channels have a different structure than our typical Channel object
+    // so we need to grab the channel off of the raw payload
+    ...rawPendingOpenChannels.map(chan => chan.channel)
+  ]
+
+  const closedChannels = [
+    ...rawClosedChannels,
+    // Pending channels have a different structure than our typical Channel object
+    // so we need to grab the channel off of the raw payload
+    ...rawPendingClosingChannels.map(chan => chan.channel),
+    ...rawPendingForceClosingChannels.map(chan => chan.channel),
+    ...rawWaitingCloseChannels.map(chan => chan.channel)
+  ]
+
+  const pendingChannels = [
+    ...rawPendingOpenChannels.map(chan => chan.channel),
+    ...rawPendingClosingChannels.map(chan => chan.channel),
+    ...rawPendingForceClosingChannels.map(chan => chan.channel),
+    ...rawWaitingCloseChannels.map(chan => chan.channel)
+  ]
+
+  const [
+    closedChannelTxIds,
+    closedChannelFees,
+    openChannelTxIds,
+    pendingChannelTxIds
+  ] = await Promise.all([
+    getTxIdsForChannels(closedChannels),
+    getFeesForClosedChannels(rawClosedChannels),
+    getTxIdsForChannels(openChannels),
+    getTxIdsForChannels(pendingChannels)
+  ])
+
+  return transactions.map(({ txHash, amount, blockHeight, timeStamp, totalFees }) => {
+    const isClosedChannel = closedChannelTxIds.has(txHash)
+    const isChannel = (openChannelTxIds.has(txHash) || closedChannelTxIds.has(txHash))
+    const isPendingChannel = pendingChannelTxIds.has(txHash)
+
+    const type = determineTransactionType(amount, blockHeight, isChannel, isClosedChannel, isPendingChannel)
+
+    let fees = Big(totalFees).div(this.quantumsPerCommon)
+
+    // If the type of transaction is a close, `totalFees` will be 0, so we must
+    // calculate the fee based on information from the closing transaction
+    if (type === TRANSACTION_TYPES.CHANNEL_CLOSE) {
+      fees = Big(closedChannelFees.get(txHash)).div(this.quantumsPerCommon)
+    }
+
+    return {
+      type,
+      amount: Big(amount).div(this.quantumsPerCommon).toString(),
+      transaction: txHash,
+      blockNumber: blockHeight,
```

Comment:
why `blockNumber` instead of `blockHeight`?

Reply:
@treygriffith no reason. Im cool changing to blockHeight